sdk/python/kubeflow/training/api/training_client.py

```diff
@@ -139,64 +139,50 @@ def train(
 
         namespace = namespace or self.namespace
 
-        if isinstance(resources_per_worker, dict):
```

Comment:
@andreyvelich  how are these validations stopping the user from running the training on cpus?

Reply:
I was wrong. I isn't stopping user of running `train` API on CPUs, but we validate if `cpu` and `memory` is set in the `resources_per_worker` parameter. Which might be not required.
E.g. user can only specify number of GPUs in `resources_per_worker`.

Reply:
@andreyvelich then shall we change the default value of resources_per_worker as it is None currently, what if the user passes an empty dict

Reply:
@deepanker13 If will be fine if user passes the empty dict, since Kubernetes will assign the resources automatically.
E.g. this works for me:
```
cl.create_job(
    resources_per_worker={},
    name="test-empty",
    num_workers=1,
    base_image="docker.io/hello-world",
)
```

As I said, if we understand that we need additional validation in the future, we can always do it in a separate PRs.

Reply:
Sure